pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java

```diff
@@ -275,7 +276,8 @@ public CompletableFuture<ClientCnx> getConnection(InetSocketAddress logicalAddre
             }
             // Try use exists connection.
             if (clientCnx.getIdleState().tryMarkUsingAndClearIdleTime()) {
-                return CompletableFuture.completedFuture(clientCnx);
+                return CompletableFuture.completedFuture(clientCnx).supplyAsync(
```

Comment:
there's no need for the `completedFuture(clientCnx)` part, that should be removed completely.